src/rars/util/Indenter.java

```diff
@@ -0,0 +1,109 @@
+package rars.util;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Basic indenter for RISC-V assembly code.
+ * Aligns labels, instructions/directives, and comments in fixed columns.
+ * Formatting rules:
+ * - Labels start at column 0
+ * - Instructions start at column 20
+ * - Comments start at column 40
+ */
+public class Indenter {
+    // Column positions for alignment
+    private static final int LABEL_COLUMN = 0;         // Column for labels
+    private static final int INSTRUCTION_COLUMN = 20;  // Column for instructions
+    private static final int COMMENT_COLUMN = 40;      // Column for comments
+
+    /**
+     * Indents a list of assembly code lines according to RISC-V conventions.
+     * @param lines List of raw assembly code lines
+     * @return List of properly indented lines
+     */
+    public List<String> indent(List<String> lines) {
+        List<String> indentedLines = new ArrayList<>();
+        
+        for (String line : lines) {
+            String trimmedLine = line.trim();
+            
+            // Handle empty lines and comments/directives
+            if (trimmedLine.isEmpty() || trimmedLine.startsWith("#")) {
+                indentedLines.add(line);
+                continue;
+            }
+
+            // Handle directives
+            if (trimmedLine.startsWith(".")) {
+                StringBuilder formattedLine = new StringBuilder();
+                formattedLine.append(trimmedLine);
+                indentedLines.add(formattedLine.toString());
+                continue;
```

Comment:
directives should be aligner like instructions.

```
        bar a1
        .baz 42
```

This is important for structures for instance

```
foo:    .word 42  # bla
        .word 124 # bla
        .dword -5 # bla
```